### PR TITLE
[full-ci] Unify people sharing sidebar (no panels, inline sharee editing)

### LINKED ISSUE #5216: When a user shares folder to a group, the sharer is shown as group in group member's sharing-sidebar
### Steps to reproduce
1. Create user `Alice` and `Brian`
2. Create group `Group101` and add user `Brian` to this group
3. Login as `Alice`, create a folder `SHAREME` and share it with group `Group101`
(the collaborator list seems fine)
4. Login as `Brian`, open sharing-sidebar (user `Alice` is displayed as group)

Alice's view (Sharer):
gs01

Brian's view (Group member aka Sharee):
gs02

### Expected behaviour
The sharer should be displayed as user (showing as group is confusing)

### Actual behaviour
The sharer is shown as a group

### Server configuration
**Operating system**: `Ubuntu 20.04`
**Web server:** `Apache`
**Database:** `MySQL`
**PHP version:** `7.4`
**ownCloud version:** `10.8.0 prealpha (git)`

### LINKED ISSUE #5299: no resharing popup displayed if a share is only reshared once
### Steps to reproduce
1. as `admin` share to `user1`
2. as `user1` share to `user2`
3. as `admin` click the `shared by user1` button in the people list

### Expected behaviour
a box should be displayed with the information

### Actual behaviour
a slightly visible empty box is displayed
it works fine if the share is reshared again from `user2` to `user3`

https://user-images.githubusercontent.com/2425577/122516599-ad547b00-d02e-11eb-982a-c9fa833910f7.mp4

I'm currently writing unit tests for that ares, lets see if I can catch the bug there

### LINKED ISSUE #5608: [Bug] "Share" does not show sharing-dialog in right sidebar
### Steps to reproduce
1. login to https://ocis.ocis-web.latest.owncloud.works/
2. create fresh folder
3. rightclick -> "share"
4. no sharing in right sidebar

### Expected behaviour
You should see the sharing dialog:
image

### LINKED ISSUE #5763: Show sharees as full list
## Acceptance Criteria
- [x] possibility to show displayname + email e.g. like: Tobias Baader (tbaader@owncloud.com)
- [X] truncate name if it gets too long
- [ ] display whole name in a tooltip if truncated
- [x] **wording** ~~invited people~~ display: "Shared with"
- [x] display additional info (share type, expiration date) below sharee name
- [x] don't show the `via` tag and the owner/resharer info in this list - they get moved to the file details in another ticket
- N.a. **resend invite:** implement only, if endpoint is already available. if not, dont implement it.
- [x] context menu:
    - [x] change role on click
    - [x] change custom rights in sub-menu
    - [x] edit expiration date in submenu
        - [x] can't add/edit expiration date after share has been created
    - [x] remove share on click, no confirmation dialog needed
    - [x] when creating share with custom rights (e.g. view&delete), delete permissions don't show up after page reload (possibly oCIS-only, needs investigation if backend/frontend issue)
- [ ] accepatance test: coordinate with JanKariTech
- [ ] (check if keyboard navigation is possible)

### LINKED ISSUE #5797: Sidebar "People": Unite into 1 panel
## Acceptance Criteria
- No "Add People" Button in main sharing panel
- instead of add people button: sharing form-panel is always directly visible
- No "Cancel" Button (other than in the screenhot)
- Only "Share" Button is visible
- "Shared with" is visible and can be expanded/collapsed

#### With collapsed list
image

#### With expanded list
image

### LINKED ISSUE #5923: Show sharing compact view in Details
This is a detail, but I leave the ticket here to start a discussion.

To me it doesn't make much sense that when I click "People" (in the sidebar) to see who I shared a file/folder with, I see first a compact view which I need to click to view the full details. This is an extra step that doesn't bring much to my experience.

I would suggest having this compact view in the Details pane, leaving the People's pane solely with the full details view.

image

Ofc that you might already have other plans for these views and therefore my suggestion might not apply.

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description
Move invite form into the people sharing main panel and introduce inline editing for existing shares (instead of an `edit` subpanel).

## Related Issue
- Fixes #5923
- Fixes #5608
- Fixes #5763
- Fixes #5797
- Fixes #5216
- Fixes #5299 (by killing the reshare information from the list entirely)

## Motivation and Context
See linked issues

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Technical debt
- [X] Tests

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
- [x] release ODS with https://github.com/owncloud/owncloud-design-system/pull/1798 being merged, update ODS in this PR (needed for padding size in OcDrop)
- [x] reload bitmaskToRole mapping for the sharee ListItem after updating a share (i.e. show `read+share` permissions as `Viewer`, not as `Custom permissions`)
- [x] fix that updating a role clears the expiration date
- [x] fix inline expiration date modifications flakyness
- [x] remove tests which are not applicable anymore (mostly the fact the we showed the `owner` if the sharee list before and don't do that anymore in this PR intentionally)
- [ ] implement todo unit tests in RoleDropdown.spec.js
- [ ] fix unit tests in FileShares.spec.js @pascalwengerter  
- [ ] implement todo unit tests in InviteCollaboratorForm.spec.js
- [ ] implement todo unit tests in RecipientContainer.spec.js
- [ ] implement todo unit tests in ExpirationDatepicker.spec.js
- [ ] fix integration tests @kulmann 
- [ ] fix acceptance tests

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/20395/16/1
The following scenarios passed on retry:
- webUISharingAcceptSharesToRoot/acceptShares.feature:108

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/20395/17/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:15

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublicExpireAndRoles</strong> https://drone.owncloud.com/owncloud/web/20409/38/1
The following scenarios passed on retry:
- webUISharingPublicExpire/shareByPublicLinkExpiringLinks.feature:11

File: packages/web-app-files/src/components/SideBar/Shares/NewCollaborator/Form.vue [hunk 1]
```diff
@@ -86,15 +71,17 @@ import { roleToBitmask } from '../../../../helpers/collaborators'
...
 
 import AutocompleteItem from './AutocompleteItem.vue'
-import CollaboratorsEditOptions from './CollaboratorsEditOptions.vue'
+import RoleSelection from '../RoleSelection.vue'
 import RecipientContainer from './RecipientContainer.vue'
+import ExpirationDatepicker from './ExpirationDatepicker.vue'
 
 export default {
-  name: 'NewCollaborator',
+  name: 'NewCollaboratorForm',
```
Comment (paulnebr) [inline packages/web-app-files/src/components/SideBar/Shares/NewCollaborator/Form.vue:79]: I like NewCollaboratorForm, but should we name the file accordingly?

Comment (pascalwengerter) [reply packages/web-app-files/src/components/SideBar/Shares/NewCollaborator/Form.vue:79]: Sure, I can take care of that! Any other suggestions regarding naming? I'm not 100% happy with what I've got so far

Comment (paulnebr) [reply packages/web-app-files/src/components/SideBar/Shares/NewCollaborator/Form.vue:79]: Maybe AddNewCollaboratorForm?

Comment (kulmann) [reply packages/web-app-files/src/components/SideBar/Shares/NewCollaborator/Form.vue:79]: ... or `InviteCollaboratorForm`?

Comment (paulnebr) [reply packages/web-app-files/src/components/SideBar/Shares/NewCollaborator/Form.vue:79]: yes that sounds good, I would also rename the folder from NewCollaborator to InviteCollaborator sounds more intuitive

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalGroups</strong> https://drone.owncloud.com/owncloud/web/20480/26/1
The following scenarios passed on retry:
- webUISharingInternalGroups/shareWithGroups.feature:166

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/20495/16/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:15

File: packages/web-app-files/tests/unit/components/SideBar/Shares/RoleDropdown.spec.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+            existingRole: PeopleShareRoles.list(type === 'folder')[0]
+          })
+          expect(wrapper).toMatchSnapshot()
+        }
+      )
+      it.todo(
+        'displays a dropdown with viewer, editor and custom permissions if no custom permissions had been selected'
+      )
+      it.todo('displays a dropdown with custom permissions if they already were selected')
+      //   it('opens custom permissions drop when custom permissions item in the roles is selected', async () => {
```
Comment (lookacat) [inline packages/web-app-files/tests/unit/components/SideBar/Shares/RoleDropdown.spec.js:134]: Can this out-commented step be removed?

File: packages/web-app-files/tests/unit/components/SideBar/Shares/RoleDropdown.spec.js [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+            await wrapper
+              .find(selectors.roleButtonPrefix + PeopleShareRoles.custom(isFolder).name)
+              .trigger('click')
+
+            expect(showHideMock).toHaveBeenCalled()
+          }
+        )
+      })
+
+      //   it.each(['update', 'create', 'delete', 'share'])(
```
Comment (lookacat) [inline packages/web-app-files/tests/unit/components/SideBar/Shares/RoleDropdown.spec.js:81]: Can this be removed?

File: packages/web-app-files/tests/unit/components/SideBar/Shares/InviteCollaborator/InviteCollaboratorForm.spec.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+describe('InviteCollaboratorForm', () => {
+  describe('renders correctly', () => {
+    it.todo('renders a select field for share receivers')
```
Comment (lookacat) [inline packages/web-app-files/tests/unit/components/SideBar/Shares/InviteCollaborator/InviteCollaboratorForm.spec.js:3]: im fine with todos, just checking if its a leftover or if we will fix this later?

File: packages/web-app-files/src/components/SideBar/Shares/InviteCollaborator/InviteCollaboratorForm.vue [hunk 5]
```diff
@@ -1,77 +1,67 @@
...
+      ref="ocSharingAutocomplete"
+      v-model="selectedCollaborators"
+      :options="autocompleteResults"
+      :loading="searchInProgress"
+      :multiple="true"
+      :filter="filterRecipients"
+      :label="selectedCollaboratorsLabel"
+      aria-describedby="files-share-invite-hint"
+      :dropdown-should-open="
+        ({ open, search }) => open && search.length >= minSearchLength && !searchInProgress
```
Comment (lookacat) [inline packages/web-app-files/src/components/SideBar/Shares/InviteCollaborator/InviteCollaboratorForm.vue:14]: i think a computed would be nice for easier reading :)

File: packages/web-app-files/tests/unit/components/SideBar/Shares/RoleDropdown.spec.js (hunk 2 above)
Comment (pascalwengerter) [reply packages/web-app-files/tests/unit/components/SideBar/Shares/RoleDropdown.spec.js:134]: Nope, to be implemented..

File: packages/web-app-files/tests/unit/components/SideBar/Shares/RoleDropdown.spec.js (hunk 3 above)
Comment (pascalwengerter) [reply packages/web-app-files/tests/unit/components/SideBar/Shares/RoleDropdown.spec.js:81]: To be implemented

File: packages/web-app-files/tests/unit/components/SideBar/Shares/InviteCollaborator/InviteCollaboratorForm.spec.js (hunk 4 above)
Comment (pascalwengerter) [reply packages/web-app-files/tests/unit/components/SideBar/Shares/InviteCollaborator/InviteCollaboratorForm.spec.js:3]: Fix later;)

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/21207/15/1
The following scenarios passed on retry:
- webUISharingAcceptSharesToRoot/acceptShares.feature:108

**Review by lookacat — APPROVED** [review]
Lets merge 🥇

**Pull request merged**